tpcx_bb/queries/q27/tpcx_bb_query_hf_27.py

```diff
@@ -0,0 +1,161 @@
+    )
+    return product_reviews_df
+
+
+def run_single_part_workflow(df, model_path):
+    """
+    This function runs the entire ner workflow end2end on a single GPU
+    """
+    import cudf
+
+    w_st = time.time()
+
+    worker = get_worker()
+    if hasattr(worker, "q27_model"):
+        model = worker.q27_model
+    else:
+        model = load_model(model_path)
+        worker.q27_model = model
+
+    id2vocab, vocab2id = create_vocab_table(os.path.join(model_path, "vocab.txt"))
+    vocab_hash_file = os.path.join(model_path, "vocab-hash.txt")
+
+    token_d, prediction_d = run_inference_on_df(
+        df, model, vocab_hash_file, batchsize=128
+    )
+
+    output_d = {}
+
+    for seq, pred_label in prediction_d.items():
+        if pred_label is not None:
+            sen_df = get_review_sentence(
+                token_d[seq], prediction_d[seq], vocab2id, id2vocab
+            )
+            review_df = token_d[seq]["df"][["pr_review_sk", "pr_item_sk"]]
+            review_df = review_df.reset_index(drop=False)
+            review_df.rename(columns={"index": "input_text_index"}, inplace=True)
+            output_d[seq] = sen_df.merge(review_df)[
+                ["pr_review_sk", "pr_item_sk", "company_name", "review_sentence"]
+            ]
+
+    del token_d, prediction_d
+
+    output_df = cudf.concat([o_df for o_df in output_d.values()])
+    output_df.rename(
+        columns={"pr_review_sk": "review_sk", "pr_item_sk": "item_sk"}, inplace=True
+    )
+
+    w_et = time.time()
+    logging.warning("Single part took = {}".format(w_et - w_st))
+    return output_df.drop_duplicates()
+
+
+def main(client, config):
+
+    import cudf
+
+    model_path = os.path.join(config["data_dir"], "../../q27_model_dir")
+    product_reviews_df = benchmark(
+        read_tables,
+        config=config,
+        compute_result=config["get_read_time"],
+        dask_profile=config["dask_profile"],
+    )
+    product_reviews_df = product_reviews_df[
+        product_reviews_df.pr_item_sk == q27_pr_item_sk
+    ].persist()
+
+    meta_d = {
+        "review_sk": np.ones(1, dtype=np.int64),
+        "item_sk": np.ones(1, dtype=np.int64),
+        "company_name": "",
+        "review_sentence": "",
+    }
+    meta_df = cudf.DataFrame(meta_d)
+    output_df = product_reviews_df.map_partitions(
+        run_single_part_workflow, model_path, meta=meta_df
+    )
+    output_df = output_df.persist()
+    wait(output_df)
+    client.run(del_model_attribute)
+    return output_df
+
+
+if __name__ == "__main__":
+    from xbb_tools.cluster_startup import attach_to_cluster
+    import cudf
+    import dask_cudf
+
+    config = tpcxbb_argparser()
+    client, bc = attach_to_cluster(config)
+    client.run(rmm.reinitialize, pool_allocator=True, initial_pool_size=14e9)
+    for i in range(0, 5):
+        st = time.time()
+        run_query(config=config, client=client, query_func=main)
+        et = time.time()
+        print(f"time taken = {et-st} s")
```

Comment:
This loop is for debugging/timing, right? I think we may be able to run this once

Reply:
Resolved by https://github.com/rapidsai/tpcx-bb/pull/84/commits/8c17699b5b182626d336745d6b449d0bfdff393d